nixos/modules/services/misc/gitea.nix

```diff
@@ -163,6 +163,58 @@ in
         };
       };
 
+      captcha = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enables Gitea to display a CAPTCHA challenge on registration.
+          '';
+        };
+
+        secretFile = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "/var/lib/secrets/gitea/captcha_secret";
+          description = "Path to a file containing the CAPTCHA secret key.";
+        };
+
+        siteKey = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "my_site_key";
+          description = "CAPTCHA site key to use for Gitea.";
+        };
+
+        url = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "https://google.com/recaptcha";
+          description = "CAPTCHA url to use for Gitea. Only relevant for `recaptcha` and `mcaptcha`.";
+        };
+
+        type = mkOption {
+          type = types.enum [ "image" "recaptcha" "hcaptcha" "mcaptcha" "cfturnstile" ];
+          default = "image";
+          example = "recaptcha";
+          description = "The type of CAPTCHA to use for Gitea.";
+        };
+
+        requireForLogin = mkOption {
+          type = types.bool;
+          default = false;
+          example = true;
+          description = "Displays a CAPTCHA challenge whenever a user logs in.";
+        };
+
+        requireForExternalRegistration = mkOption {
+          type = types.bool;
+          default = false;
```

Comment:
This follows upstreams defaults, right?

Reply:
Yes, this value follows the default value for `REQUIRE_EXTERNAL_REGISTRATION_PASSWORD`. Any non-`nullOr` value follows Gitea's configuration defaults: https://docs.gitea.com/next/administration/config-cheat-sheet#service-service